src/main/java/roomescape/reservation/external/toss/TossApiClient.java

```diff
@@ -0,0 +1,55 @@
+package roomescape.reservation.external.toss;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.RestClientResponseException;
+import roomescape.global.exception.ErrorCode;
+import roomescape.global.exception.ExternalApiException;
+
+@Component
+public class TossApiClient {
+
+    private final ObjectMapper objectMapper;
+    private final RestClient tossRestClient;
+
+    public TossApiClient(final ObjectMapper objectMapper,
+                         final @Qualifier("tossRestClient") RestClient tossRestClient) {
+        this.objectMapper = objectMapper;
+        this.tossRestClient = tossRestClient;
+    }
+
+    public TossPaymentResponse requestPayment(TossPaymentRequest tossPaymentRequest) {
+        try {
+            return tossRestClient.post()
+                    .uri("/payments/confirm")
+                    .body(objectMapper.convertValue(tossPaymentRequest, Map.class))
```

Comment:
RestClient 의 체인닝 메서드 중 하나인 body가 메서드 파라미터로 어떤 타입을 수용하는지 확인해보면 좋을 것 같아요. 😃

<img width="986" alt="image" src="https://github.com/user-attachments/assets/27fca2a7-e021-417a-81ba-de324cc98730" />

```suggestion
                    .body(tossPaymentRequest)
```

Reply:
수정 완료했습니다 👍🏻